CMakeLists.txt

```diff
@@ -98,6 +98,13 @@ option(IREE_BUILD_EXPERIMENTAL_PYTHON_GENERATED_BENCHMARKS "Builds IREE benchmar
 # information on how to declare external drivers.
 set(IREE_EXTERNAL_HAL_DRIVERS "" CACHE STRING "")
 
+# Additional executable loader deps to add dependent libraries to any target
+# using the default executable loader registration utilities.
+# TODO(benvanik): extend the deps to encompass the built-in loaders too so that
+# we have one flag. We could also support a list of deps and automatically
+# generate the registration from that via a configure file.
+set(IREE_HAL_EXECUTABLE_LOADER_DEPS "" CACHE STRING "")
```

Comment:
Maybe put "EXTRA" or "ADDITIONAL" in the name? I'm also weary of how large this root CMakeLists.txt file is getting. Maybe we should extract more implementation details into other files, so user-configurable variables/options like this one are easier to read.